File: daa/structures/test/com/tw/vrc/lists/StackTest.java
package com.tw.vrc.lists;

import org.junit.Test;

import static junit.framework.Assert.assertEquals;

public class StackTest {

    @Test
    public void shouldWork() {
        Stack<String> stack = new Stack<String>();
        stack.push("A");
        stack.push("B");
        stack.push("C");
        assertEquals("C",stack.pop());
        assertEquals("B",stack.pop());
        assertEquals("A",stack.pop());
    }
}

File: algos/graphs/src/com/vijayrc/algos/stack/StackTest.scala
package com.vijayrc.algos.stack

import org.scalatest.FunSuite

class StackTest extends FunSuite{
  test("should push/pop to link based stack"){
    val stack:Stack = new StackLinks
    stack.push("1").push("3").pop()
    stack.push("4").push("2")
    stack.show()
  }
  test("should push/pop to array based stack"){
    val stack:Stack = new StackArray(2)
    stack.push("1").push("3").pop()
    stack.push("4").push("2").push("5").push("6")
    stack.show()
    (1 to 10).foreach(x=>stack.pop())
    stack.show()
  }
  test("should evaluate simple expressions"){
    val dijkstra = new Dijkstra2Stack
    val evaluate: Int = dijkstra.evaluate("(((2*3)*(2+6))-(4*3))")
    assert(36 == evaluate)
  }

}
